TestModels/SharedMakefile.mk

```diff
@@ -297,5 +351,94 @@ _clean:
+	rm -rf $(LIBRARY_ROOT)/runtimes/python/src/**/internal_generated_dafny/*.py
+	rm -rf $(LIBRARY_ROOT)/runtimes/python/src/**/smithygenerated
+	rm -rf $(LIBRARY_ROOT)/runtimes/python/test/internal_generated_dafny/*.py
+	rm -rf $(LIBRARY_ROOT)/runtimes/python/.tox
+	rm -rf $(LIBRARY_ROOT)/runtimes/python/poetry.lock
+	rm -rf $(LIBRARY_ROOT)/runtimes/python/build
+
+########################## Python targets
+
+build_python: _python_underscore_dependency_extern_names
+build_python: _python_underscore_extern_names
+build_python: transpile_dependencies_python
+build_python: build_implementation_python
+build_python: transpile_test_python
+build_python: _python_revert_underscore_extern_names
+build_python: _python_revert_underscore_dependency_extern_names
+build_python: _mv_internal_generated_dafny_python
+build_python: _remove_src_module_python
+
+build_implementation_python: TARGET=py
+build_implementation_python: OUT=runtimes/python/dafny_src
+build_implementation_python: COMPILE_SUFFIX_OPTION=
+build_implementation_python: build_implementation
+
+# `transpile_implementation_python` is not directly used, but is indirectly used via `transpile_dependencies`
+# The `transpile` target does NOT include the Dafny runtime library (_dafny.py) in the generated code
+# while the `build` target does
+transpile_implementation_python: transpile_dependencies_python
+transpile_implementation_python: transpile_src_python
+transpile_implementation_python: transpile_test_python
+transpile_implementation_python: _mv_internal_generated_dafny_python
+transpile_implementation_python: _remove_src_module_python
+
+transpile_src_python: TARGET=py
+transpile_src_python: OUT=runtimes/python/dafny_src
+transpile_src_python: COMPILE_SUFFIX_OPTION=
+transpile_src_python: transpile_implementation
+
+transpile_test_python: TARGET=py
+transpile_test_python: OUT=runtimes/python/test
+transpile_test_python: COMPILE_SUFFIX_OPTION=
+transpile_test_python: transpile_test
+
+# Hacky workaround until Dafny supports per-language extern names.
+# Replaces `.`s with `_`s in strings like `{:extern ".*"}`.
+# This is flawed logic and should be removed, but is a reasonable band-aid for now.
+# TODO: Once Dafny supports per-language extern names, remove and replace with Pythonic extern names.
+# This is tracked in https://github.com/dafny-lang/dafny/issues/4322.
+# This may require new Smithy-Dafny logic to generate Pythonic extern names.
+_python_underscore_extern_names:
+	find src -regex ".*\.dfy" -type f -exec sed -i $(SED_PARAMETER) '/.*{:extern \".*\".*/s/\./_/g' {} \;
+	find Model -regex ".*\.dfy" -type f -exec sed -i $(SED_PARAMETER) '/.*{:extern \".*\.*"/s/\./_/g' {} \;
+	find test -regex ".*\.dfy" -type f -exec sed -i $(SED_PARAMETER) '/.*{:extern \".*\".*/s/\./_/g' {} \;
+
+_python_underscore_dependency_extern_names:
+	$(MAKE) -C $(STANDARD_LIBRARY_PATH) _python_underscore_extern_names
+	$(patsubst %, $(MAKE) -C $(PROJECT_ROOT)/% _python_underscore_extern_names;, $(LIBRARIES))
```

Comment:
Praise/Note: 
Nice way to "Make it Work". 
This is very close to solution written by an ARG Intern from 2019.